core/include/seeding/track_params_estimation_helper.hpp

```diff
@@ -0,0 +1,135 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include "edm/seed.hpp"
+#include "edm/spacepoint.hpp"
+#include "edm/track_parameters.hpp"
+
+namespace traccc {
+
+/// helper functions (for both cpu and gpu) to calculate bound track parameter
+/// at the bottom spacepoint
+///
+/// @param seed is the input seed
+/// @param bfield is the magnetic field
+/// @param mass is the mass of particle
+inline TRACCC_HOST_DEVICE bound_vector seed_to_bound_vector(
+    const seed& seed, const vector3& bfield, const scalar mass) {
+
+    bound_vector params;
+
+    const auto& spB = seed.spB;
+    const auto& spM = seed.spM;
+    const auto& spT = seed.spT;
+
+    darray<vector3, 3> sp_global_positions;
+    sp_global_positions[0] = spB.global;
+    sp_global_positions[1] = spM.global;
+    sp_global_positions[2] = spT.global;
+
+    // Define a new coordinate frame with its origin at the bottom space
+    // point, z axis long the magnetic field direction and y axis
+    // perpendicular to vector from the bottom to middle space point.
+    // Hence, the projection of the middle space point on the tranverse
+    // plane will be located at the x axis of the new frame.
+    vector3 relVec = sp_global_positions[1] - sp_global_positions[0];
+    vector3 newZAxis = vector::normalize(bfield);
+    vector3 newYAxis = vector::normalize(vector::cross(newZAxis, relVec));
+    vector3 newXAxis = vector::cross(newYAxis, newZAxis);
+
+    // The center of the new frame is at the bottom space point
+    vector3 translation = sp_global_positions[0];
+
+    transform3 trans(translation, newZAxis, newXAxis);
+
+    // The coordinate of the middle and top space point in the new frame
+    auto local1 = trans.point_to_local(sp_global_positions[1]);
+    auto local2 = trans.point_to_local(sp_global_positions[2]);
+
+    // printf("%f %f %f %f %f %f \n", local1[0], local1[1], local1[2],
+    // local2[0], local2[1], local2[2]);
+
+    // Lambda to transform the coordinates to the (u, v) space
+    auto uvTransform = [](const vector3& local) -> vector2 {
+        vector2 uv;
+        scalar denominator = local[0] * local[0] + local[1] * local[1];
+        uv[0] = local[0] / denominator;
+        uv[1] = local[1] / denominator;
+        return uv;
+    };
```

Comment:
This function doesn't actually act as a closure - maybe we can just define it as a free function? Or maybe move it the algebra plugins?

Reply:
Hmm I don't recommend moving it into algebra plugin.
I will just make a free function in the same file